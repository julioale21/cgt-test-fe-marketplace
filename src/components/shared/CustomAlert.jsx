import React from 'react';
import PropTypes from 'prop-types';
import { Snackbar, Alert } from '@mui/material';

export const CustomAlert = ({
  open,
  message,
  severity = 'success',
  onClose,
  autoHideDuration = 3000
}) => {
  return (
    <Snackbar
      open={open}
      autoHideDuration={autoHideDuration}
      onClose={onClose}
      anchorOrigin={{ vertical: 'bottom', horizontal: 'left' }}
    >
      <Alert onClose={onClose} severity={severity} variant="filled" sx={{ width: '100%' }}>
        {message}
      </Alert>
    </Snackbar>
  );
};

CustomAlert.propTypes = {
  open: PropTypes.bool.isRequired,
  message: PropTypes.string.isRequired,
  severity: PropTypes.oneOf(['error', 'warning', 'info', 'success']),
  onClose: PropTypes.func.isRequired,
  autoHideDuration: PropTypes.number
};

CustomAlert.defaultProps = {
  severity: 'success',
  autoHideDuration: 3000
};
