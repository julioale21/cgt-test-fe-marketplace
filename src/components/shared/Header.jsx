import { Link } from 'react-router-dom';
import ShoppingCartIcon from '@mui/icons-material/ShoppingCart';
import { AppBar, Toolbar, Typography, Badge, Container } from '@mui/material';
import PropTypes from 'prop-types';

const Header = ({ cartItemsCount = 0 }) => {
  return (
    <AppBar position="static">
      <Container maxWidth="lg" disableGutters>
        <Toolbar sx={{ display: 'flex', justifyContent: 'space-between', px: { xs: 2, sm: 3 } }}>
          <Typography
            variant="h6"
            component={Link}
            to="/"
            sx={{ textDecoration: 'none', color: 'inherit' }}
          >
            Marketplace
          </Typography>
          <div>
            <Link to="/" style={{ textDecoration: 'none', color: 'inherit', marginRight: '20px' }}>
              Home
            </Link>
            <Link to="/cart" style={{ textDecoration: 'none', color: 'inherit' }}>
              <Badge badgeContent={cartItemsCount} color="error" max={99}>
                <ShoppingCartIcon />
              </Badge>
            </Link>
          </div>
        </Toolbar>
      </Container>
    </AppBar>
  );
};

Header.propTypes = {
  cartItemsCount: PropTypes.number
};

export { Header };
