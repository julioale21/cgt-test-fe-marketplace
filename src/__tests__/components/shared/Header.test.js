// src/components/shared/Header/Header.test.jsx
import { render, screen } from '@testing-library/react';
import { Header } from '../../../components/shared/Header';

describe('Header', () => {
  const renderHeader = (props = {}) => {
    return render(<Header {...props} />);
  };

  it('renders marketplace title', () => {
    renderHeader();
    const title = screen.getByText('Marketplace');
    expect(title).toBeInTheDocument();
    expect(title.tagName.toLowerCase()).toBe('a');
    expect(title).toHaveAttribute('href', '/');
  });

  it('renders navigation links', () => {
    renderHeader();
    const homeLink = screen.getByText('Home');
    expect(homeLink).toBeInTheDocument();
    expect(homeLink).toHaveAttribute('href', '/');
  });

  it('displays cart with correct item count', () => {
    renderHeader({ cartItemsCount: 5 });
    const badge = screen.getByText('5');
    expect(badge).toBeInTheDocument();
    expect(badge.closest('a')).toHaveAttribute('href', '/cart');
  });


  it('displays 99+ when cart items exceed 99', () => {
    renderHeader({ cartItemsCount: 100 });
    const badge = screen.getByText('99+');
    expect(badge).toBeInTheDocument();
    expect(badge.closest('a')).toHaveAttribute('href', '/cart');
  });
});