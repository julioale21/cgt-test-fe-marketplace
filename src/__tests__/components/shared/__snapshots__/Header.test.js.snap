// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot with cart items 1`] = `
<div>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-ptdwpq-MuiPaper-root-MuiAppBar-root"
    style="--Paper-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-disableGutters css-ho17r4-MuiContainer-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1n5sh1q-MuiToolbar-root"
      >
        <a
          class="MuiTypography-root MuiTypography-h6 css-jwpxwx-MuiTypography-root"
          href="/"
        >
          Marketplace
        </a>
        <div>
          <a
            href="/"
            style="text-decoration: none; margin-right: 20px;"
          >
            Home
          </a>
          <a
            href="/cart"
            style="text-decoration: none;"
          >
            <span
              class="MuiBadge-root css-63ezxx-MuiBadge-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                data-testid="ShoppingCartIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2M1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2"
                />
              </svg>
              <span
                class="MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorError css-1enwswh-MuiBadge-badge"
              >
                5
              </span>
            </span>
          </a>
        </div>
      </div>
    </div>
  </header>
</div>
`;

exports[`Header matches snapshot with default props 1`] = `
<div>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-ptdwpq-MuiPaper-root-MuiAppBar-root"
    style="--Paper-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-disableGutters css-ho17r4-MuiContainer-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1n5sh1q-MuiToolbar-root"
      >
        <a
          class="MuiTypography-root MuiTypography-h6 css-jwpxwx-MuiTypography-root"
          href="/"
        >
          Marketplace
        </a>
        <div>
          <a
            href="/"
            style="text-decoration: none; margin-right: 20px;"
          >
            Home
          </a>
          <a
            href="/cart"
            style="text-decoration: none;"
          >
            <span
              class="MuiBadge-root css-63ezxx-MuiBadge-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                data-testid="ShoppingCartIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2M1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2"
                />
              </svg>
              <span
                class="MuiBadge-badge MuiBadge-standard MuiBadge-invisible MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorError css-nsw3vh-MuiBadge-badge"
              />
            </span>
          </a>
        </div>
      </div>
    </div>
  </header>
</div>
`;

exports[`Header matches snapshot with max cart items 1`] = `
<div>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-ptdwpq-MuiPaper-root-MuiAppBar-root"
    style="--Paper-shadow: 0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12);"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg MuiContainer-disableGutters css-ho17r4-MuiContainer-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1n5sh1q-MuiToolbar-root"
      >
        <a
          class="MuiTypography-root MuiTypography-h6 css-jwpxwx-MuiTypography-root"
          href="/"
        >
          Marketplace
        </a>
        <div>
          <a
            href="/"
            style="text-decoration: none; margin-right: 20px;"
          >
            Home
          </a>
          <a
            href="/cart"
            style="text-decoration: none;"
          >
            <span
              class="MuiBadge-root css-63ezxx-MuiBadge-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1umw9bq-MuiSvgIcon-root"
                data-testid="ShoppingCartIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7 18c-1.1 0-1.99.9-1.99 2S5.9 22 7 22s2-.9 2-2-.9-2-2-2M1 2v2h2l3.6 7.59-1.35 2.45c-.16.28-.25.61-.25.96 0 1.1.9 2 2 2h12v-2H7.42c-.14 0-.25-.11-.25-.25l.03-.12.9-1.63h7.45c.75 0 1.41-.41 1.75-1.03l3.58-6.49c.08-.14.12-.31.12-.48 0-.55-.45-1-1-1H5.21l-.94-2zm16 16c-1.1 0-1.99.9-1.99 2s.89 2 1.99 2 2-.9 2-2-.9-2-2-2"
                />
              </svg>
              <span
                class="MuiBadge-badge MuiBadge-standard MuiBadge-anchorOriginTopRight MuiBadge-anchorOriginTopRightRectangular MuiBadge-overlapRectangular MuiBadge-colorError css-1enwswh-MuiBadge-badge"
              >
                99+
              </span>
            </span>
          </a>
        </div>
      </div>
    </div>
  </header>
</div>
`;
